leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/QueueModeIntegrationTestHelper.java

```diff
@@ -0,0 +1,350 @@
+                Type.OPAQUE, null, null, null);
+        ResourceModel objlnkfield = new ResourceModel(OBJLNK_MULTI_INSTANCE_RESOURCE_ID, "objlnk", Operations.RW, true,
+                false, Type.OBJLNK, null, null, null);
+        ResourceModel objlnkSinglefield = new ResourceModel(OBJLNK_SINGLE_INSTANCE_RESOURCE_ID, "objlnk", Operations.RW,
+                false, false, Type.OBJLNK, null, null, null);
+        objectModels.add(new ObjectModel(TEST_OBJECT_ID, "testobject", null, ObjectModel.DEFAULT_VERSION, false, false,
+                stringfield, booleanfield, integerfield, floatfield, timefield, opaquefield, objlnkfield,
+                objlnkSinglefield));
+
+        return objectModels;
+    }
+
+    public void initialize() {
+        currentEndpointIdentifier.set("leshan_integration_test_" + r.nextInt());
+    }
+
+    public String getCurrentEndpoint() {
+        return currentEndpointIdentifier.get();
+    }
+
+    public void createClient() {
+        // Create objects Enabler
+        ObjectsInitializer initializer = new ObjectsInitializer(new LwM2mModel(createObjectModels()));
+        initializer.setInstancesForObject(LwM2mId.SECURITY, Security.noSec(
+                "coap://" + server.getUnsecuredAddress().getHostString() + ":" + server.getUnsecuredAddress().getPort(),
+                12345));
+        initializer.setInstancesForObject(LwM2mId.SERVER, new Server(12345, LIFETIME, BindingMode.UQ, false));
+        initializer.setInstancesForObject(LwM2mId.DEVICE, new Device("Eclipse Leshan", MODEL_NUMBER, "12345", "UQ") {
+            @Override
+            public ExecuteResponse execute(int resourceid, String params) {
+                if (resourceid == 4) {
+                    return ExecuteResponse.success();
+                } else {
+                    return super.execute(resourceid, params);
+                }
+            }
+        });
+        List<LwM2mObjectEnabler> objects = initializer.createMandatory();
+        objects.addAll(initializer.create(2, 2000));
+
+        // Build Client
+        LeshanClientBuilder builder = new LeshanClientBuilder(currentEndpointIdentifier.get());
+        builder.setObjects(objects);
+        client = builder.build();
+    }
+
+    public void createServer(int clientAwakeTime) {
+        server = createServerBuilder(clientAwakeTime).build();
+        server.getPresenceService().addListener(new PresenceListener() {
+
+            @Override
+            public void onAwake(Registration registration) {
+                if (registration.getEndpoint().equals(currentEndpointIdentifier.get())) {
+                    awakeNotifications++;
+                }
+
+            }
+
+            @Override
+            public void onSleeping(Registration registration) {
+                awakeNotifications = 0;
+                awakeLatch.countDown();
+                scheduleUpdateAfterSleeping((int) SLEEPTIME);
```

Comment:
I'm not sure this is really needed.
You can use directly triggerUpdate in tests? (this is maybe a way to speed up the tests)

Reply:
I was trying to make a more realistic tests, where the client actually "sleep" for some time. But yeah, I guess we can also trigger the update message, simulating a "0 sleep time", that will test the server functionality. Would you prefer that? 

We can also maintain this way, and the user can set the sleeptime to 0 if it wants this behavior. I would prefer this option.

Reply:
It works better